server/data/strapiApiTypes.ts

```diff
@@ -1,52 +1,70 @@
 import { components } from '../@types/strapi-api'
 
+export type ListResponse<T> = {
+  data?: T[]
+  meta?: {
+    pagination?: {
+      page?: number
+      pageCount?: number
+      pageSize?: number
+      total?: number
+    }
+  }
+}
```

Comment:
Defined this generic type so that we can hopefully (🤞 ) tweak the `T` type without having to unwrap and wrap types.  

Maybe something like this
```ts
type HasTeam = { team?: Team }
export type Product = components['schemas']['Product'] & HasTeam
  ```

And we won't need to tweak all the calling sites

Reply:
General idea being that we don't do any manual changing of the type file and instead do any patching to fix it here.. which should hopefully help us limp on until we migrate strapi versions.

I think in lots of places we should unwrap the page response in the strapi client and just return raw arrays - I don't think we really use the pagination in lots of places